lib/routes/hitsz/due-tzgg.ts

```diff
@@ -0,0 +1,165 @@
+
+    const baseUrl = 'http://due.hitsz.edu.cn';
+
+    // 根据 id 构建基础路径
+    const baseListPath = `index/tzggqb`;
+
+    // --- 步骤 1: 获取所有需要抓取的页面URL ---
+    const pageUrls = [];
+    // 第一页 URL
+    pageUrls.push(new URL(`${baseListPath}.htm`, baseUrl).href);
+
+    // 1. 修复：显式初始化变量为 null，指定类型（避免隐式 undefined）
+    let firstPageResponse: Awaited<ReturnType<typeof got>> | null = null;
+    // 获取后续页面的 URL
+    try {
+        firstPageResponse = await got(pageUrls[0]);
+        // 2. 修复：首次访问 data 前增加空值检查（核心防错逻辑）
+        if (!firstPageResponse) {
+            throw new Error('Failed to get valid response for first page');
+        }
+        const $ = load(firstPageResponse.data);
+
+        // 解析页面上的翻页链接
+        const pageLinks = new Set();
+
+        // 查找所有分页链接
+        $('.p_no a, .p_next a, .p_last a').each((_, el) => {
+            const href = $(el).attr('href');
+            if (href) {
+                // 确保相对路径正确转换为完整URL
+                const fullHref = href.startsWith('http') ? href : `index/${href}`;
+                const fullUrl = new URL(fullHref, baseUrl).href;
+                pageLinks.add(fullUrl);
+            }
+        });
+
+        // 添加找到的页面链接
+        const additionalUrls = [...pageLinks];
+        for (let i = 0; i < Math.min(fetchPageCount - 1, additionalUrls.length); i++) {
+            pageUrls.push(additionalUrls[i]);
+        }
+    } catch {
+        // 捕获错误时，firstPageResponse 保持为 null（无需额外赋值）
+    }
+
+    // --- 步骤 2: 并发抓取所有页面的列表 ---
+    const pagePromises = pageUrls.map((url) => got(url).catch(() => null));
+    const pageResponses = await Promise.all(pagePromises);
+
+    const detailPromises = [];
+
+    for (const response of pageResponses) {
+        if (!response) {
+            continue;
+        } // 忽略失败的请求
+
+        const $ = load(response.data);
+        // 使用原来的选择器
+        const listItemsOnPage = $('ul.list-main-modular li, .list-main-modular-text-list li').toArray();
+
+        // 修改：将 for 循环 push 改为 map 生成数组（PR评论要求：Use map instead of push）
+        const pageItems = listItemsOnPage
+            .map((el) => {
+                const $el = $(el);
+                const linkUrl = $el.find('a').attr('href');
+                if (!linkUrl) {
+                    return null;
+                } // 过滤无链接项
+
+                const title = $el.find('span.text-over, a').text().trim();
+                const pubDateStr = $el.find('label').text().trim();
+
+                return {
+                    title,
+                    link: new URL(linkUrl, baseUrl).href,
+                    pubDate: pubDateStr ? timezone(parseDate(pubDateStr), 8) : null,
+                    description: title, // 使用标题作为描述
+                };
+            })
+            .filter(Boolean); // 过滤 null 项
+
+        detailPromises.push(...pageItems); // 展开数组添加到承诺列表
+    }
+
+    // --- 步骤 3: 并发抓取所有文章的详细内容 ---
+    const allResolvedItems = (await Promise.all(detailPromises)).filter(Boolean);
```

Comment:
https://github.com/DIYgod/RSSHub/blob/b998b5ad9e79035a2eacd5c35716eabe69e79067/lib/routes/hitsz/due-tzgg.ts#L59-L92 are synchronous which does not require Promise.all. Please use [flatMap](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap) instead.